ui/app/src/shared-worker.ts

```diff
@@ -107,6 +107,7 @@ function unauthenticated(excludeClient?: MessagePort) {
   log(`Auth has expired.`);
   clearTimeout(timeout);
   status = 'expired';
+  socketClient?.deactivate();
```

Comment:
What was missing was to deactivate the client when the session expired. That caused the client to retry the connection causing the 302 error. When re-connecting, the client already gets enabled again.